frappe/integrations/doctype/slack_webhook_url/slack_webhook_url.py

```diff
@@ -40,7 +42,7 @@ def send_slack_message(webhook_url, message, reference_doctype, reference_name):
 			]
 		}
 	]
-	data = {"text": message, "attachments": attachments}
+	data = {"text": message, "attachments": attachments if show_link else []}
```

Comment:
Seems like Lines 31-44 aren't required if `show_link` is False. Maybe add an if-block before this to set `attachments`'s value instead of this?